src/platform/Linux/PlatformManagerImpl.cpp

```diff
@@ -108,13 +108,13 @@ gboolean WiFiIPChangeListener(GIOChannel * ch, GIOCondition /* condition */, voi
                             continue;
                         }
 
-                        if (ConnectivityMgrImpl().GetWiFiIfName() == nullptr)
+                        if (ConnectivityMgrImpl().GetWiFiIfName() == nullptr && ConnectivityMgrImpl().GetEthernetIfName() == nullptr)
                         {
-                            ChipLogDetail(DeviceLayer, "No wifi interface name. Ignoring IP update event.");
+                            ChipLogDetail(DeviceLayer, "No wifi interface name or etherInterface name. Ignoring IP update event.");
                             continue;
                         }
 
-                        if (strcmp(name, ConnectivityMgrImpl().GetWiFiIfName()) != 0)
+                        if (strcmp(name, ConnectivityMgrImpl().GetWiFiIfName()) != 0 && strcmp(name, ConnectivityMgrImpl().GetEthernetIfName()) != 0)
```

Comment:
This might be passing null to strcmp, no?  Because the test above will pass if _either_ pointer is non-null.

This needs to be more carefully written.

Reply:
new code will be:

                        bool wifi_matched = connectivityMgrImpl().GetWiFiIfName()!= nullptr && strcmp(name, ConnectivityMgrImpl().GetWiFiIfName()) == 0;
                        bool ethernet_matched = ConnectivityMgrImpl().GetEthernetIfName() != nullptr && strcmp(name, ConnectivityMgrImpl().GetEthernetIfName()) == 0;
                        if(!wifi_matched && !ethernet_matched){
                             ChipLogDetail(DeviceLayer, "No wifi interface name or ethernet interface name match the router info interface name . Ignoring IP update event.");
                            continue;
                        }